packages/financial-templates-lib/src/price-feed/DefiPulseTotalPriceFeed.js

```diff
@@ -0,0 +1,145 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const assert = require("assert");
+
+// An implementation of PriceFeedInterface that uses DefiPulse Data api to retrieve prices.
+class DefiPulseTotalPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs the DefiPulseTVL_ALLPriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from truffle instance to connect to Ethereum network.
+   * @param {String} apiKey DeFiPulse Data API key. Note: these API keys are rate-limited.
+   * @param {Integer} lookback How far in the past the historical prices will be available using getHistoricalPrice.
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the current time.
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   */
+  constructor(logger, web3, apiKey, lookback, networker, getTime, minTimeBetweenUpdates) {
+    super();
+    this.logger = logger;
+    this.web3 = web3;
+
+    this.apiKey = apiKey;
+    assert(apiKey, "requires apiKey for Defi Pulse");
+
+    this.lookback = lookback;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+
+    this.toWei = this.web3.utils.toWei;
+
+    this.historicalPrices = [];
+  }
+
+  getCurrentPrice() {
+    return this.currentPrice;
+  }
+
+  getHistoricalPrice(time) {
+    if (this.lastUpdateTime === undefined) {
+      return undefined;
+    }
+
+    let closestTime = { timestamp: 0, tvlUSD: 0 };
+
+    // Go through all values and find time that that is the largest and still less than 'time'
+    for (let i = 0; i < this.historicalPrices.length; i++) {
+      let past = this.historicalPrices[i].timestamp;
+      let val = this.historicalPrices[i].tvlUSD;
+
+      if (past > closestTime.timestamp && past < time) {
+        closestTime.timestamp = past;
+        closestTime.tvlUSD = val;
+      }
+    }
+
+    const historicalPrice = this.scaleResult(closestTime.tvlUSD);
+
+    if (closestTime.timestamp === 0) {
+      return undefined;
+    } else {
+      return historicalPrice;
+    }
+  }
+
+  getLastUpdateTime() {
+    return this.lastUpdateTime;
+  }
+
+  async update() {
+    const currentTime = this.getTime();
+
+    // Return early if the last call was too recent.
+    if (this.lastUpdateTime !== undefined && this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime) {
+      this.logger.debug({
+        at: "DEFI_PULSE_TOTAL_TVL_PriceFeed",
```

Comment:
nit: you should make this `at` the same as the file name. I know this feed was previous all upper case like this but that's no longer the case.
```suggestion
        at: "DefiPulseTotalPriceFeed",
```

Reply:
Changed.